sky/packages/sky/lib/src/material/bottom_sheet.dart

```diff
@@ -101,9 +133,17 @@ class _ModalBottomSheetRoute extends PerformanceRoute {
     );
   }
 
+  void didPush(NavigatorState navigator) {
+    super.didPush(navigator);
+    _performance?.forward();
+  }
+
   void didPop([dynamic result]) {
     completer.complete(result);
-    super.didPop(result);
+    if (_performance.status != PerformanceStatus.dismissed) {
+      _performance?.reverse();
+      super.didPop(result);
```

Comment:
I'm not sure what is right, but I'm pretty sure this is wrong. I think it'll leave the route in an inconsistent state. What were you trying to do here?